spellbook/models/seaport/ethereum/seaport_ethereum_transfers.sql

```diff
@@ -0,0 +1,822 @@
+-- ## this function consists of 4 parts,
+-- ## p1_ : seaport."Seaport_call_fulfillBasicOrder"
+-- ## p2_ : seaport."Seaport_call_fulfillAvailableAdvancedOrders"
+-- ##       seaport."Seaport_call_fulfillAvailableOrders"
+-- ## p3_ : seaport."Seaport_call_fulfillOrder"
+-- ##       seaport."Seaport_call_fulfillAdvancedOrder"
+-- ## p4_ : seaport."Seaport_call_matchOrders"
+-- ##     : seaport."Seaport_call_matchAdvancedOrders"
+
+{{ config(
+        alias ='transfers',
+        materialized ='incremental',
+        file_format ='delta',
+        incremental_strategy='merge',
+        unique_key='unique_trade_id'
+        )
+}}
+
+with p1_call as (
+    select 'basic_order' as main_type
+          ,call_tx_hash as tx_hash
+          ,call_block_time as block_time
+          ,call_block_number as block_number
+          ,max(get_json_object(parameters, "$.basicOrderType")) as order_type_id
+      from {{ source('seaport_ethereum','Seaport_call_fulfillBasicOrder') }}
+      {% if is_incremental() %} -- this filter will only be applied on an incremental run 
+      where call_block_time >= (select max(block_time) from {{ this }}) 
+      {% endif %}
+     group by 1,2,3,4
+)
+
+,p1_evt as (
+    select c.main_type
+          ,c.tx_hash
+          ,c.block_time
+          ,c.block_number
+          ,c.order_type_id
+          ,'offer' as sub_type
+          ,offer_idx as sub_idx
+          ,e.offerer as sender
+          ,e.recipient as receiver
+          ,e.zone
+          ,concat('0x',substr(get_json_object(offer2, "$.token"),3,40)) as token_contract_address 
+          ,get_json_object(offer2, "$.amount") as original_amount
+          ,get_json_object(offer2, "$.itemType") as item_type
+          ,get_json_object(offer2, "$.identifier") as token_id
+          ,e.contract_address as exchange_contract_address
+          ,e.evt_index
+      from 
+          (select *, posexplode(offer) as (offer_idx, offer2) from {{ source('seaport_ethereum','Seaport_evt_OrderFulfilled') }}) e 
+          inner join p1_call c on c.tx_hash = e.evt_tx_hash
+      {% if is_incremental() %} -- this filter will only be applied on an incremental run 
```

Comment:
this incremental logic should be applied in the 'e' subquery in line 50 rather than here, since you already filter 'c' alias table in prior CTE. the 'e' alias is reading from a source again. placing it within the () above on line 50 should work